e-library/src/Components/forms/createbook.jsx

```diff
@@ -1,29 +1,73 @@
-import React from "react";
+import React, { useState } from "react";
+import { useNavigate } from 'react-router-dom';
+import axios from "../api/axios";
+import formData from 'form-data'
+import { userInfo } from "../homepage/body";
 
-const CreateBook = () => {
-    return(
-        <form >
-        <h2>Create your book</h2>
-        <div className="input-box">
-            <input type="text" placeholder="Title"/>
-        </div>
-        <div className="input-box">
-           <textarea placeholder="Description" rows="4" columns="50"></textarea>
-        </div>
-        <div className="input-box">
-            <input type="text" placeholder="Author"/>
-        </div>
-        <div className="input-box">
-            <input type="text" placeholder="Price"/>
-        </div>
-        <div className="input-box">
-            <input type="file"/>
-        </div>
-        <div className="input-box">
-            <input type="submit" value="Create your book"/>
-        </div>
-    </form>
-    )
-}
+export const bookData = {};
+
+export const CreateBook = () => {
+    const [book, setBook] = useState({
+        title: '',
+        price: '',
+        author: '',
+        description: '',
+        fileUrl: '',
+    })
 
-export default CreateBook
\ No newline at end of file
+    const redirect = useNavigate()
+
+    const submit = async (e) => {
+        e.preventDefault()
+        const form = new formData()
+        form.append('title', book.title)
+        form.append('price', book.price)
+        form.append('author', book.author)
+        form.append('description', book.description)
+        form.append('bookFile', book.fileUrl)
+        try {
+            const response = await axios.post('/api/books', form, {
+                headers: {
+                    'Content-Type': 'multipart/form-data',
+                    'authorization': userInfo.token,
+                }
+            });
+            setBook({ title: ''});
+            setBook({ price: ''});
+            setBook({ author: ''});
+            setBook({ fileUrl: ''});
+            setBook({ description: ''})
+            window.alert('Book has been created successfully')
+            redirect('/homepage')
+            bookData.img = response.data.createbook.imageUrl
+            bookData.file = response.data.createbook.fileUrl
+        } catch (err){
+              if (err) {
+                  window.alert('Failed to create book. Please try again')
+                  redirect('/homepage')
+                  console.log(err)
+            }
+        }
+    }
+        return(
+            <>      
+               <div> 
+                    <form onSubmit={submit}>
+                        <h2>Create your book</h2><div className="input-box">
+                            <input type="text" value={book.title} placeholder="Title" onChange={(e) => setBook({ ...book, title: e.target.value})} />
```

Comment:
move the logic in the onchange function to a custom hook so you just have `onChange={onChange}`